File: src/offer11.Min_number_in_roatedarray/min_number_in_roatedarray.cpp
class Solution {
public:
    int minNumberInRotateArray(vector<int> rotateArray) {
        int ll = 0;
        int rr = rotateArray.size()-1;
        int mid = ll;   //！！！如果数组没有进行旋转
        while(rotateArray[ll] >= rotateArray[rr]){
            if((rr - ll) == 1){
                mid = rr;
                break;
            }
            mid = ll + (rr - ll) / 2;
            if(rotateArray[mid] < rotateArray[ll]){
                rr = mid;
            }
            else{
                ll = mid;
            }
        }
        return rotateArray[rr];
    }
};

File: src/offer11-1.Min_number_in_roatedarray/min_number_in_roatedarray.java
import java.util.ArrayList;
public class Solution {
    public int minNumberInRotateArray(int [] array) {
        if(array == null || array.length == 0){
            return 0;
        }
        int left = 0;
        int right = array.length - 1;
        while(left < right){
            int mid = left + (right - left)/2;
            if(array[mid] < array[right]){
                right = mid;
            } else{
                left = mid + 1;
            }
        }
        return array[left];
    }
}